src/cpp/shared_core/include/shared_core/FilePath.hpp

```diff
@@ -231,6 +260,53 @@ class FilePath
     */
    static Error uniqueFilePath(const std::string& in_basePath, const std::string& in_extension, FilePath& out_filePath);
 
+#ifndef _WIN32
+   /**
+    * @brief Changes the file mode to the specified file mode.
+    *
+    * @param in_fileModeStr     The posix file mode string. e.g. rwxr-xr-x.
+    *
+    * @return Success if the file mode could be changed; Error otherwise.
+    */
+   Error changeFileMode(const std::string& in_fileModeStr) const;
+
+   /**
+    * @brief Changes the file mode to the specified file mode.
+    *
+    * @param in_fileMode        The new file mode.
+    * @param in_setStickyBit    Whether to set the sticky bit on this file.
+    *
+    * @return Success if the file mode could be changed; Error otherwise.
+    */
+   Error changeFileMode(FileMode in_fileMode, bool in_setStickyBit = false) const;
+
+   /**
+    * @brief Changes the ownership of the file or directory to the specified user.
+    *
+    * @param in_newUser         The user who should own the file.
+    * @param in_recursive       If this FilePath is a directory, whether to recursively change ownership on all files
+    *                           and directories within this directory.
+    * @param in_shouldChown     A recursive iteration function which allows the caller to filter files and directories.
```

Comment:
minor nit: the lines here are just long enough to force GitHub's PR UI to overflow horizontally a bit; e.g.

![Screen Shot 2020-02-03 at 10 38 33 AM](https://user-images.githubusercontent.com/1976582/73680600-63edd080-4671-11ea-8652-06b525a85363.png)

Not sure if this is motivation enough for us to adjust our editor line widths down a bit.

Reply:
80? Did someone say 80?! ;-)